src/table/boot.rs

```diff
@@ -238,6 +253,40 @@ pub struct BootServices {
 }
 
 impl BootServices {
+    /// Get the [`Handle`] of the currently-executing image.
+    pub fn image_handle(&self) -> Handle {
+        // Safety:
+        //
+        // `IMAGE_HANDLE` is only set by `set_image_handle`, see that
+        // documentation for more details.
+        //
+        // Additionally, `image_handle` takes a `&self` which ensures it
+        // can only be called while boot services are active. (After
+        // exiting boot services, the image handle should not be
+        // considered valid.)
+        unsafe { IMAGE_HANDLE.handle.get().read().unwrap() }
```

Comment:
Maybe we should `expect` with a message indicating that `set_image_handle` hasn't been called beforehand? (although it's going to be a rare error, since most people probably use `#[entry]`).

Reply:
done